cluster/shard_writer.go

```diff
@@ -43,7 +43,9 @@ func (w *ShardWriter) WriteShard(shardID, ownerID uint64, points []tsdb.Point) e
 	if !ok {
 		panic("wrong connection type")
 	}
-	defer conn.Close() // return to pool
+	defer func(conn net.Conn) {
```

Comment:
No.  Fixes the connection not getting marked as unusable. Without it, the connection still seems be marked a usable even after it's marked as unusable.  cc/ @corylanou